src/hipo2root.c

```diff
@@ -12,20 +17,31 @@
 #include "../lib/bank_containers.h"
 
 int main(int argc, char **argv) {
-    char *in_filename = NULL;
-    int  run_no = -1;
-
-    if (hipo2root_handle_args_err(hipo2root_handle_args(argc, argv, &in_filename, &run_no),
-                                  &in_filename))
-        return 1;
+    char *in_filename  = NULL;
+    char *out_filename = NULL;
 
-    char *out_filename = (char *) malloc(22 * sizeof(char));
-    if      (run_no /     10 == 0) sprintf(out_filename, "../root_io/00000%d.root", run_no);
-    else if (run_no /    100 == 0) sprintf(out_filename, "../root_io/0000%d.root", run_no);
-    else if (run_no /   1000 == 0) sprintf(out_filename, "../root_io/000%d.root", run_no);
-    else if (run_no /  10000 == 0) sprintf(out_filename, "../root_io/00%d.root", run_no);
-    else if (run_no / 100000 == 0) sprintf(out_filename, "../root_io/0%d.root", run_no);
-    else                           sprintf(out_filename, "../root_io/%d.root", run_no);
+    int  run_no    = -1;
+    bool use_simul = false;
+    
+    if (hipo2root_handle_args_err(hipo2root_handle_args(argc, argv, &in_filename, &run_no, &use_simul),
+                                      &in_filename))
+            return 1;
```

Comment:
This is just me being nitpicky, but why did you change the indentation here? Please follow the program's convention of 4 spaces, even if only to maintain my sanity.

Reply:
My IDE went fuckin' nuts. Solved it.